drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -410,7 +410,16 @@ NTSTATUS TCPConnect
     Status = TCPTranslateError(LibTCPConnect(Connection,
                                                 &connaddr,
                                                 RemotePort));
-
+    if (Status != STATUS_SUCCESS && Status != STATUS_PENDING)
+    { 
+    	TI_DbgPrint(DEBUG_TCP,
+                    ("No pcb from LibTCPConnect, clear ConnectRequest\n")); 
+    	LockObject(Connection);
+    	RemoveTailList(&Connection->ConnectRequest); 
```

Comment:
`RemoveEntryList(&Bucket->Entry)` would be more appropriate. Since the connection was unlocked after the insert call, another item might now be at the tail position.